### PR TITLE
Update dependencies 

### PR DESCRIPTION
**Pull request by mdiazmel**
Update the dependencies for Clinica Packages

- [x] Remove unnecessary requirements for Pypi package (`requirements.txt`).
- [x] Remove unnecessary requirements for Conda environment (`environment.yml`).
- [x] Add `requirements-dev.txt` file
- [x] Update files to have minimal versions for all packages (and not fixed versions).
- [x] Test CI, instantiation test.
- [x] Test CI,  non-regression tests.

This PR addresses issue #87

### CODE REVIEW

**Comment by mdiazmel** [comment]
Contributions from @alexandreroutier:

- Use MRtrix3.Tractography() from new Nipype version (contains @jguillon's PR)
- Fix import issue for DWIPreprocessing (niflow-nipype1-workflows and scikit-image were needed)
-  Update ANTSRegistration interface connections for DWIPreprocessing

Todolist:
- [x] Make dwi-preprocessing-using-fieldmap work
- [x] Make dwi-preprocessing-using-t1 work
- [x] Make dwi-connectome work
- [ ] Check that Similarity() function works
- [ ] CI dwi-preprocessing-using-fieldmap
- [ ] CI dwi-preprocessing-using-t1
- [ ] CI dwi-connectome

**Comment by mdiazmel** [comment]
Actual environment uses this third-parties :

- ANTs 2.2.0
- ITK 4.13
- Freesurfer 6.0.0
- PETPVC 1.2.2
- MRtrix3 3.0.RC3
- FSL 5.0.11
- SPM12 r7487

File: clinica/iotools/utils/pipeline_handling.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def pet_volume_pipeline(caps_dir, df, **kwargs):
                             atlas_df = pd.read_csv(atlas_path, sep='\t')
                             label_list = [group + '_' + atlas + '_ROI-' + str(x) for x in
                                           range(n_regions)]
-                            ses_df.loc[:, label_list] = atlas_df['mean_scalar'].as_matrix()
+                            ses_df.loc[:, label_list] = atlas_df['mean_scalar'].values
```
Comment (alexandreroutier) [inline clinica/iotools/utils/pipeline_handling.py:176]: According to [pandas API](https://pandas.pydata.org/pandas-docs/version/0.25.1/reference/api/pandas.DataFrame.values.html#pandas.DataFrame.values), they recommend to use `.to_numpy()` instead.

File: test/nonregression/testing_tools.py [hunk 2]
```diff
@@ -35,10 +35,10 @@ def likeliness_measure(file1, file2, threshold1, threshold2, display=False):
     import matplotlib.pyplot as plt
 
     print(' ** comparing ' + os.path.basename(file1) + ' **')
-    data1 = nib.load(file1).get_data()
+    data1 = nib.load(file1).get_fdata()
```
Comment (alexandreroutier) [inline test/nonregression/testing_tools.py:38]: Todo for later: replace other `get_data()` by `get_fdata()` (cf [doc](https://nipy.org/nibabel/reference/nibabel.dataobj_images.html))

Comment (mdiazmel) [reply test/nonregression/testing_tools.py:38]: Hi @alexandreroutier ,
I don't understand why you pointed this line. The change in line 38 already done!

**Comment by mdiazmel** [comment]
Done! I'll merge.

File: test/nonregression/testing_tools.py (hunk 2 above)
Comment (alexandreroutier) [reply test/nonregression/testing_tools.py:38]: Sorry for the misunderstanding.

The `get_data()` methods are also present in Clinica ( warnings are silenced when running Clinica) so we will have to take care of them one day :)

**Pull request merged**